src/components/SignupPopup/SignupPopup.js

```diff
@@ -15,16 +16,32 @@ function SignupPopup({ ...props }) {
     usePopups().popups.signUpPopup;
   const { openSignInPopup } = usePopups().popups.signInPopup;
 
+  const { onRegister, authServerError, changeAuthError } = useAuth();
+
+  function handleClose() {
+    closeSignUpPopup();
+    changeAuthError('');
+    resetForm();
+  }
+
+  function handleSubmit() {
+    const userData = { email, password, name };
+    onRegister(userData);
+  }
+
   return (
     <>
```

Comment:
* `<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component